h2o-algos/src/main/java/hex/schemas/GAMV3.java

```diff
@@ -228,20 +231,35 @@
     @API(help = "Number of knots for gam predictors", level = Level.critical, gridable = true)
     public int[] num_knots;
 
-    @API(help = "Predictor column names for gam", required = true, level = Level.critical, gridable = true)
-    public String[] gam_columns;
+/*    @API(help = "Number of eigenvectors to use in SVD for thin plate regression.  If specified, must be of the same" +
+            " length as gam_columns even if the bs type is not thin-plate regression with SVD.", 
+            level = Level.critical, gridable = true)
+    public int[] k;*/ // for later version
 
-    @API(help = "Smoothing parameter for gam predictors", level = Level.critical, gridable = true)
+    @API(help = "Arrays of predictor column names for gam for smoothers using single or multiple predictors like " +
+            "{{'c1'},{'c2','c3'},{'c4'},...}", required = true, level = Level.critical, gridable = true)
+    public String[][] gam_columns;
+
+    @API(help = "Smoothing parameter for gam predictors.  If specified, must be of the same length as gam_columns",
+            level = Level.critical, gridable = true)
     public double[] scale;
 
-    @API(help = "Basis function type for each gam predictors, 0 for cr", level = Level.critical, gridable = true)
+    @API(help = "Basis function type for each gam predictors, 0 for cr, 1 for thin plate regression with knots, 2 for" +
+            " thin plate regression with SVD.  If specified, must be the same size as gam_columns",
+            level = Level.critical, gridable = true)
     public int[] bs;
     //public BSType bs;
 
     @API(help="Save keys of model matrix", level = Level.secondary, direction = Direction.INPUT)
     public boolean keep_gam_cols; // if true will save keys storing GAM columns
+
+    @API(help="standardize TP predictor columns", level = Level.secondary, direction = Direction.INPUT)
```

Comment:
Please, explain what the TP shortcut is in the documentation (if it is not too complicated).

Reply:
No problem.

Reply:
Done